Octree: Add `layers` property for filtering objects.

RemusMar opened the pull request:
This is a very useful feature (especially in games), it's 100% backwards compatible with the exisiting users logic and I thought to share it.
Long story short: many times you need to ignore a list of models/meshes when performing collisions or raycasting.
Think about the vegetation in a 3D world for example.
And here is this feature in action: https://necromanthus.com/Test/html5/SMC_Octree.html

RemusMar committed:
Octree and ignore certain meshes

Mugen87 commented:
Would you be okay if we implement this differently via layers similar to `Raycaster`? Meaning:

- Adding a new `layers` property to `Octree`.
- Make a layer test in `fromGraphNode()`.
```
if ( this.layers.test( object.layers ) ) {

    // add to octree

}
```
If you shift objects to different layers, they should be excluded from the addition.

RemusMar commented:
Hi Michael,
Sure, but I thought the current solution is the easiest and the fastest.

Also, I hope you noticed the "per material name" instead of "per mesh name" approach.
I think the "per material name" is the best solution, at least for 2 reasons:
1) think about 100 bushes sharing the same material.
You want to add a single entry in that array, not 100
2) think about trees: you want collisions for the tree bark, but not for the upper part.

Mugen87 commented:
Yeah, that makes sense. But if the class works with layers, the app can use whatever criteria to configure the layers setting.

RemusMar committed:
Octree with Layers

RemusMar commented:
OK.
I've added Layers to Octree and updated the sample file: https://necromanthus.com/Test/html5/SMC_Octree.html 
Same good result with only difference that the filtering is executed on the "client side" now:

```
	if (["palm1","palm2","palm3","bush1","bush2","bush3"].includes(child.material.name) ) {
		child.layers.set( 1 );
	}
// ...
camera.layers.enable( 1 );
```
